code/app/src/main/java/com/cmput301w21t06/crowdfly/Database/CommentController.java

```diff
@@ -0,0 +1,181 @@
+package com.cmput301w21t06.crowdfly.Database;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.cmput301w21t06.crowdfly.Controllers.ExperimentLog;
+import com.cmput301w21t06.crowdfly.Models.Comment;
+import com.cmput301w21t06.crowdfly.Models.Experiment;
+import com.cmput301w21t06.crowdfly.Models.User;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.EventListener;
+import com.google.firebase.firestore.FirebaseFirestoreException;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This class controls all operations related to Trials
+ */
+public class CommentController {
+    private CollectionReference cCollection;
+    private ArrayList<Comment> comments = new ArrayList<Comment>();
+    private String qID;
+    private String eID;
+
+    public CommentController(String eid, String qID) {
+        this.qID = qID;
+        this.eID = eid;
+        cCollection = GodController.getDb().collection(CrowdFlyFirestorePaths.comments(eid, qID));
+        setUp();
+    }
+    /**
+     * This sets up the snapshot listener for trials

```

Comment:
do you mean this sets up the snapshot listener for comments?

Reply:
I fixed these comments and javadocs, i'll be posting a pr for this soon!